contracts/modules/TransferManager/GTM/GeneralTransferManager.sol

```diff
@@ -188,6 +190,22 @@ contract GeneralTransferManager is GeneralTransferManagerStorage, TransferManage
         return (Result.NA, bytes32(0));
     }
 
+    function _isValidModule(address _holder) internal view returns(bool) {
+        uint8[] memory types;
+        bool isArchived;
+        bytes32 name;
+        // Use getModule() function to avoid the malicious module change its factory type after adding
+        // into the securityToken 
+        (name,,,isArchived,types,)= securityToken.getModule(_holder);
+        if (name != bytes32(0) && !isArchived) {
```

Comment:
Why do we care about `name != bytes32(0)`?

Reply:
The idea was to avoid going into the loop when the given address is not the module.

Reply:
Probably better to check `module` returned value, rather than `name` then I think.